compose/service.py

```diff
@@ -925,7 +925,7 @@ def get_container_data_volumes(container, volumes_option):
             continue
 
         # Copy existing volume from old container
-        volume = volume._replace(external=mount['Source'])
+        volume = volume._replace(external=mount['Name'])
```

Comment:
It should, I believe named volumes were in 1.9. Our test suite still runs against 1.9, so it should confirm for us.